--- ../src-base/minecraft/net/minecraft/entity/passive/EntityLlama.java
+++ ../src-work/minecraft/net/minecraft/entity/passive/EntityLlama.java
@@ -47,6 +47,8 @@
 import net.minecraft.world.World;
 import net.minecraft.world.storage.loot.LootTableList;
 
+import carpet.CarpetSettings;
+
 public class EntityLlama extends AbstractChestHorse implements IRangedAttackMob
 {
     private static final DataParameter<Integer> field_190720_bG = EntityDataManager.<Integer>func_187226_a(EntityLlama.class, DataSerializers.field_187192_b);
@@ -192,7 +194,7 @@
             j = 6;
             f = 10.0F;
 
-            if (this.func_110248_bS() && this.func_70874_b() == 0)
+            if (this.func_110248_bS() && this.func_70874_b() == 0 && !(CarpetSettings.llamaOverfeedingFix && this.func_70880_s() ) )
             {
                 flag = true;
                 this.func_146082_f(p_190678_1_);
@@ -492,6 +494,19 @@
         }
     }
 
+    /*
+     * Breeding items for llamas: CARPET-XCOM
+     */
+    @Override
+    protected boolean isValidBreedingItem(ItemStack itemstack){
+    	Item item = itemstack.func_77973_b();
+
+    	if(item == Items.field_151015_O) return false;
+    	else if(item == Item.func_150898_a(Blocks.field_150407_cf)) return false;
+
+    	return true;
+    }
+
     public boolean func_190684_dE()
     {
         return false;
@@ -513,9 +528,9 @@
                 super(p_i47285_1_, EntityWolf.class, 16, false, true, (Predicate)null);
             }
 
-            public boolean func_75250_a()
+            public boolean shouldStart()
             {
-                if (super.func_75250_a() && this.field_75309_a != null && !((EntityWolf)this.field_75309_a).func_70909_n())
+                if (super.shouldStart() && this.field_75309_a != null && !((EntityWolf)this.field_75309_a).func_70909_n())
                 {
                     return true;
                 }
@@ -539,7 +554,7 @@
                 super(p_i47282_1_, false);
             }
 
-            public boolean func_75253_b()
+            public boolean shouldContinue()
             {
                 if (this.field_75299_d instanceof EntityLlama)
                 {
@@ -552,7 +567,7 @@
                     }
                 }
 
-                return super.func_75253_b();
+                return super.shouldContinue();
             }
         }
 
