--- ../src-base/minecraft/net/minecraft/village/Village.java
+++ ../src-work/minecraft/net/minecraft/village/Village.java
@@ -73,17 +73,21 @@
 
         int i = this.field_75588_h / 10;
 
-        if (this.field_75587_j < i && this.field_75584_b.size() > 20 && this.field_75586_a.field_73012_v.nextInt(7000) == 0)
+
+        if (this.field_75587_j < i && this.field_75584_b.size() > 20)
         {
-            Vec3d vec3d = this.func_179862_a(this.field_75582_d, 2, 4, 2);
+            if (this.field_75586_a.field_73012_v.nextInt(7000) == 0) {
+                Vec3d vec3d = this.func_179862_a(this.field_75582_d, 2, 4, 2);
 
-            if (vec3d != null)
-            {
-                EntityIronGolem entityirongolem = new EntityIronGolem(this.field_75586_a);
-                entityirongolem.func_70107_b(vec3d.field_72450_a, vec3d.field_72448_b, vec3d.field_72449_c);
-                this.field_75586_a.func_72838_d(entityirongolem);
-                ++this.field_75587_j;
+                if (vec3d != null)
+                {
+                    EntityIronGolem entityirongolem = new EntityIronGolem(this.field_75586_a);
+                    entityirongolem.func_70107_b(vec3d.field_72450_a, vec3d.field_72448_b, vec3d.field_72449_c);
+                    this.field_75586_a.func_72838_d(entityirongolem);
+                    ++this.field_75587_j;
+                }
             }
+
         }
     }
 
@@ -368,6 +372,7 @@
                 flag = true;
                 villagedoorinfo.func_179853_a(true);
                 iterator.remove();
+                VillageCollection.updateMarkers = true;
             }
         }
 
@@ -379,7 +384,7 @@
 
     private boolean func_179860_f(BlockPos p_179860_1_)
     {
-        IBlockState iblockstate = this.field_75586_a.func_180495_p(p_179860_1_);
+        IBlockState iblockstate = this.field_75586_a.getBlockState(p_179860_1_, "Village loading");
         Block block = iblockstate.func_177230_c();
 
         if (block instanceof BlockDoor)
