--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIOcelotAttack.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIOcelotAttack.java
@@ -18,7 +18,7 @@
         this.func_75248_a(3);
     }
 
-    public boolean func_75250_a()
+    public boolean shouldStart()
     {
         EntityLivingBase entitylivingbase = this.field_75409_b.func_70638_az();
 
@@ -33,7 +33,7 @@
         }
     }
 
-    public boolean func_75253_b()
+    public boolean shouldContinue()
     {
         if (!this.field_75410_c.func_70089_S())
         {
@@ -45,7 +45,7 @@
         }
         else
         {
-            return !this.field_75409_b.func_70661_as().func_75500_f() || this.func_75250_a();
+            return !this.field_75409_b.func_70661_as().func_75500_f() || this.shouldStart();
         }
     }
 
@@ -55,7 +55,7 @@
         this.field_75409_b.func_70661_as().func_75499_g();
     }
 
-    public void func_75246_d()
+    public void tick()
     {
         this.field_75409_b.func_70671_ap().func_75651_a(this.field_75410_c, 30.0F, 30.0F);
         double d0 = (double)(this.field_75409_b.field_70130_N * 2.0F * this.field_75409_b.field_70130_N * 2.0F);
@@ -83,4 +83,9 @@
             }
         }
     }
+
+    @Override
+    public String getTask() {
+        return ": " + field_75409_b.func_70005_c_() + " Task: Ocelot attacking";
+    }
 }
