--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIBreakDoor.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIBreakDoor.java
@@ -15,9 +15,9 @@
         super(p_i1618_1_);
     }
 
-    public boolean func_75250_a()
+    public boolean shouldStart()
     {
-        if (!super.func_75250_a())
+        if (!super.shouldStart())
         {
             return false;
         }
@@ -38,7 +38,7 @@
         this.field_75359_i = 0;
     }
 
-    public boolean func_75253_b()
+    public boolean shouldContinue()
     {
         double d0 = this.field_75356_a.func_174818_b(this.field_179507_b);
         boolean flag;
@@ -64,9 +64,9 @@
         this.field_75356_a.field_70170_p.func_175715_c(this.field_75356_a.func_145782_y(), this.field_179507_b, -1);
     }
 
-    public void func_75246_d()
+    public void tick()
     {
-        super.func_75246_d();
+        super.tick();
 
         if (this.field_75356_a.func_70681_au().nextInt(20) == 0)
         {
@@ -89,4 +89,9 @@
             this.field_75356_a.field_70170_p.func_175718_b(2001, this.field_179507_b, Block.func_149682_b(this.field_151504_e));
         }
     }
+
+    @Override
+    public String getTask() {
+        return ": " + field_75356_a.func_70005_c_() + " Task: Breaking door";
+    }
 }
