--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIMoveIndoors.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIMoveIndoors.java
@@ -19,7 +19,7 @@
         this.func_75248_a(1);
     }
 
-    public boolean func_75250_a()
+    public boolean shouldStart()
     {
         BlockPos blockpos = new BlockPos(this.field_75424_a);
 
@@ -54,7 +54,7 @@
         }
     }
 
-    public boolean func_75253_b()
+    public boolean shouldContinue()
     {
         return !this.field_75424_a.func_70661_as().func_75500_f();
     }
@@ -88,4 +88,9 @@
         this.field_75421_d = this.field_75422_b.func_179856_e().func_177952_p();
         this.field_75422_b = null;
     }
+
+    @Override
+    public String getTask() {
+        return ": " + field_75424_a.func_70005_c_() + " Task: Moving indoors";
+    }
 }
