--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIRunAroundLikeCrazy.java
+++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIRunAroundLikeCrazy.java
@@ -49,7 +49,7 @@
         this.field_111180_a.func_70661_as().func_75492_a(this.field_111179_c, this.field_111176_d, this.field_111177_e, this.field_111178_b);
     }
 
-    public boolean func_75253_b()
+    public boolean continueExecuting()
     {
         return !this.field_111180_a.func_110248_bS() && !this.field_111180_a.func_70661_as().func_75500_f() && this.field_111180_a.func_184207_aI();
     }
@@ -84,4 +84,9 @@
             this.field_111180_a.field_70170_p.func_72960_a(this.field_111180_a, (byte)6);
         }
     }
+
+    @Override
+    public String getTask() {
+        return ": " + field_111180_a.func_70005_c_() + " Task: Run around like crazy";
+    }
 }
