--- ../src-base/minecraft/net/minecraft/entity/monster/EntityIllusionIllager.java
+++ ../src-work/minecraft/net/minecraft/entity/monster/EntityIllusionIllager.java
@@ -213,9 +213,9 @@
             super();
         }
 
-        public boolean func_75250_a()
+        public boolean shouldStart()
         {
-            if (!super.func_75250_a())
+            if (!super.shouldStart())
             {
                 return false;
             }
@@ -272,9 +272,9 @@
             super();
         }
 
-        public boolean func_75250_a()
+        public boolean shouldStart()
         {
-            if (!super.func_75250_a())
+            if (!super.shouldStart())
             {
                 return false;
             }
