--- ../src-base/minecraft/net/minecraft/entity/monster/EntitySilverfish.java
+++ ../src-work/minecraft/net/minecraft/entity/monster/EntitySilverfish.java
@@ -202,9 +202,9 @@
                 }
             }
 
-            public boolean func_75253_b()
+            public boolean continueExecuting()
             {
-                return this.field_179484_c ? false : super.func_75253_b();
+                return this.field_179484_c ? false : super.continueExecuting();
             }
 
             public void func_75249_e()
