--- ../src-base/minecraft/net/minecraft/entity/monster/EntityPigZombie.java
+++ ../src-work/minecraft/net/minecraft/entity/monster/EntityPigZombie.java
@@ -250,9 +250,9 @@
                 super(p_i45829_1_, EntityPlayer.class, true);
             }
 
-            public boolean func_75250_a()
+            public boolean shouldStart()
             {
-                return ((EntityPigZombie)this.field_75299_d).func_175457_ck() && super.func_75250_a();
+                return ((EntityPigZombie)this.field_75299_d).func_175457_ck() && super.shouldStart();
             }
         }
 }
