--- ../src-base/minecraft/net/minecraft/entity/monster/EntitySpider.java
+++ ../src-work/minecraft/net/minecraft/entity/monster/EntitySpider.java
@@ -212,7 +212,7 @@
                 super(p_i46676_1_, 1.0D, true);
             }
 
-            public boolean func_75253_b()
+            public boolean continueExecuting()
             {
                 float f = this.field_75441_b.func_70013_c();
 
@@ -223,7 +223,7 @@
                 }
                 else
                 {
-                    return super.func_75253_b();
+                    return super.continueExecuting();
                 }
             }
 
