--- ../src-base/minecraft/net/minecraft/entity/monster/EntityEnderman.java
+++ ../src-work/minecraft/net/minecraft/entity/monster/EntityEnderman.java
@@ -416,7 +416,7 @@
                 super.func_75251_c();
             }
 
-            public boolean func_75253_b()
+            public boolean continueExecuting()
             {
                 if (this.field_179448_g != null)
                 {
@@ -432,7 +432,7 @@
                 }
                 else
                 {
-                    return this.field_75309_a != null && ((EntityPlayer)this.field_75309_a).func_70089_S() ? true : super.func_75253_b();
+                    return this.field_75309_a != null && ((EntityPlayer)this.field_75309_a).func_70089_S() ? true : super.continueExecuting();
                 }
             }
 
