--- ../src-base/minecraft/net/minecraft/entity/monster/EntityVex.java
+++ ../src-work/minecraft/net/minecraft/entity/monster/EntityVex.java
@@ -253,7 +253,7 @@
             }
         }
 
-        public boolean func_75253_b()
+        public boolean continueExecuting()
         {
             return EntityVex.this.func_70605_aq().func_75640_a() && EntityVex.this.func_190647_dj() && EntityVex.this.func_70638_az() != null && EntityVex.this.func_70638_az().func_70089_S();
         }
@@ -372,7 +372,7 @@
             return !EntityVex.this.func_70605_aq().func_75640_a() && EntityVex.this.field_70146_Z.nextInt(7) == 0;
         }
 
-        public boolean func_75253_b()
+        public boolean continueExecuting()
         {
             return false;
         }
