--- ../src-base/minecraft/net/minecraft/block/BlockTripWire.java
+++ ../src-work/minecraft/net/minecraft/block/BlockTripWire.java
@@ -3,6 +3,8 @@
 import java.util.List;
 import java.util.Random;
 import javax.annotation.Nullable;
+
+import carpet.CarpetSettings;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyBool;
@@ -23,7 +25,9 @@
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
 
-public class BlockTripWire extends Block
+import redstone.multimeter.block.Meterable;
+
+public class BlockTripWire extends Block implements Meterable /* RSMM */
 {
     public static final PropertyBool field_176293_a = PropertyBool.func_177716_a("powered");
     public static final PropertyBool field_176294_M = PropertyBool.func_177716_a("attached");
@@ -146,6 +150,7 @@
     {
         if (!p_180650_1_.field_72995_K)
         {
+            if (CarpetSettings.calmITTCrashers && p_180650_1_.field_72999_e) return;
             if (((Boolean)p_180650_1_.func_180495_p(p_180650_2_).func_177229_b(field_176293_a)).booleanValue())
             {
                 this.func_176288_d(p_180650_1_, p_180650_2_);
@@ -266,4 +271,10 @@
     {
         return BlockFaceShape.UNDEFINED;
     }
+
+    // RSMM
+    @Override
+    public boolean isActive(World world, BlockPos pos, IBlockState state) {
+        return state.func_177229_b(field_176293_a);
+    }
 }
