--- ../src-base/minecraft/net/minecraft/block/BlockCactus.java
+++ ../src-work/minecraft/net/minecraft/block/BlockCactus.java
@@ -1,6 +1,9 @@
 package net.minecraft.block;
 
 import java.util.Random;
+
+import carpet.CarpetSettings;
+import carpet.helpers.HopperCounter;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyInteger;
@@ -9,7 +12,10 @@
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.Entity;
+import net.minecraft.entity.item.EntityItem;
 import net.minecraft.init.Blocks;
+import net.minecraft.item.EnumDyeColor;
+import net.minecraft.item.ItemStack;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.math.AxisAlignedBB;
@@ -52,12 +58,12 @@
                 {
                     p_180650_1_.func_175656_a(blockpos, this.func_176223_P());
                     IBlockState iblockstate = p_180650_3_.func_177226_a(field_176587_a, Integer.valueOf(0));
-                    p_180650_1_.func_180501_a(p_180650_2_, iblockstate, 4);
+                    p_180650_1_.func_180501_a(p_180650_2_, iblockstate, CarpetSettings.blockStateSyncing ? 6 : 4);
                     iblockstate.func_189546_a(p_180650_1_, blockpos, this, p_180650_2_);
                 }
                 else
                 {
-                    p_180650_1_.func_180501_a(p_180650_2_, p_180650_3_.func_177226_a(field_176587_a, Integer.valueOf(j + 1)), 4);
+                    p_180650_1_.func_180501_a(p_180650_2_, p_180650_3_.func_177226_a(field_176587_a, Integer.valueOf(j + 1)), CarpetSettings.blockStateSyncing ? 6 : 4);
                 }
             }
         }
