--- ../src-base/minecraft/net/minecraft/world/gen/feature/WorldGenTaiga2.java
+++ ../src-work/minecraft/net/minecraft/world/gen/feature/WorldGenTaiga2.java
@@ -22,21 +22,21 @@
         super(p_i2025_1_);
     }
 
-    public boolean func_180709_b(World p_180709_1_, Random p_180709_2_, BlockPos p_180709_3_)
+    public boolean generate_vanilla(World worldIn, Random rand, BlockPos position)
     {
-        int i = p_180709_2_.nextInt(4) + 6;
-        int j = 1 + p_180709_2_.nextInt(2);
+        int i = rand.nextInt(4) + 6;
+        int j = 1 + rand.nextInt(2);
         int k = i - j;
-        int l = 2 + p_180709_2_.nextInt(2);
+        int l = 2 + rand.nextInt(2);
         boolean flag = true;
 
-        if (p_180709_3_.func_177956_o() >= 1 && p_180709_3_.func_177956_o() + i + 1 <= 256)
+        if (position.func_177956_o() >= 1 && position.func_177956_o() + i + 1 <= 256)
         {
-            for (int i1 = p_180709_3_.func_177956_o(); i1 <= p_180709_3_.func_177956_o() + 1 + i && flag; ++i1)
+            for (int i1 = position.func_177956_o(); i1 <= position.func_177956_o() + 1 + i && flag; ++i1)
             {
                 int j1;
 
-                if (i1 - p_180709_3_.func_177956_o() < j)
+                if (i1 - position.func_177956_o() < j)
                 {
                     j1 = 0;
                 }
@@ -47,13 +47,13 @@
 
                 BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-                for (int k1 = p_180709_3_.func_177958_n() - j1; k1 <= p_180709_3_.func_177958_n() + j1 && flag; ++k1)
+                for (int k1 = position.func_177958_n() - j1; k1 <= position.func_177958_n() + j1 && flag; ++k1)
                 {
-                    for (int l1 = p_180709_3_.func_177952_p() - j1; l1 <= p_180709_3_.func_177952_p() + j1 && flag; ++l1)
+                    for (int l1 = position.func_177952_p() - j1; l1 <= position.func_177952_p() + j1 && flag; ++l1)
                     {
                         if (i1 >= 0 && i1 < 256)
                         {
-                            Material material = p_180709_1_.func_180495_p(blockpos$mutableblockpos.func_181079_c(k1, i1, l1)).func_185904_a();
+                            Material material = worldIn.func_180495_p(blockpos$mutableblockpos.func_181079_c(k1, i1, l1)).func_185904_a();
 
                             if (material != Material.field_151579_a && material != Material.field_151584_j)
                             {
@@ -74,34 +74,34 @@
             }
             else
             {
-                Block block = p_180709_1_.func_180495_p(p_180709_3_.func_177977_b()).func_177230_c();
+                Block block = worldIn.func_180495_p(position.func_177977_b()).func_177230_c();
 
-                if ((block == Blocks.field_150349_c || block == Blocks.field_150346_d || block == Blocks.field_150458_ak) && p_180709_3_.func_177956_o() < 256 - i - 1)
+                if ((block == Blocks.field_150349_c || block == Blocks.field_150346_d || block == Blocks.field_150458_ak) && position.func_177956_o() < 256 - i - 1)
                 {
-                    this.func_175921_a(p_180709_1_, p_180709_3_.func_177977_b());
-                    int i3 = p_180709_2_.nextInt(2);
+                    this.func_175921_a(worldIn, position.func_177977_b());
+                    int i3 = rand.nextInt(2);
                     int j3 = 1;
                     int k3 = 0;
 
                     for (int l3 = 0; l3 <= k; ++l3)
                     {
-                        int j4 = p_180709_3_.func_177956_o() + i - l3;
+                        int j4 = position.func_177956_o() + i - l3;
 
-                        for (int i2 = p_180709_3_.func_177958_n() - i3; i2 <= p_180709_3_.func_177958_n() + i3; ++i2)
+                        for (int i2 = position.func_177958_n() - i3; i2 <= position.func_177958_n() + i3; ++i2)
                         {
-                            int j2 = i2 - p_180709_3_.func_177958_n();
+                            int j2 = i2 - position.func_177958_n();
 
-                            for (int k2 = p_180709_3_.func_177952_p() - i3; k2 <= p_180709_3_.func_177952_p() + i3; ++k2)
+                            for (int k2 = position.func_177952_p() - i3; k2 <= position.func_177952_p() + i3; ++k2)
                             {
-                                int l2 = k2 - p_180709_3_.func_177952_p();
+                                int l2 = k2 - position.func_177952_p();
 
                                 if (Math.abs(j2) != i3 || Math.abs(l2) != i3 || i3 <= 0)
                                 {
                                     BlockPos blockpos = new BlockPos(i2, j4, k2);
 
-                                    if (!p_180709_1_.func_180495_p(blockpos).func_185913_b())
+                                    if (!worldIn.func_180495_p(blockpos).func_185913_b())
                                     {
-                                        this.func_175903_a(p_180709_1_, blockpos, field_181646_b);
+                                        this.func_175903_a(worldIn, blockpos, field_181646_b);
                                     }
                                 }
                             }
@@ -124,15 +124,15 @@
                         }
                     }
 
-                    int i4 = p_180709_2_.nextInt(3);
+                    int i4 = rand.nextInt(3);
 
                     for (int k4 = 0; k4 < i - i4; ++k4)
                     {
-                        Material material1 = p_180709_1_.func_180495_p(p_180709_3_.func_177981_b(k4)).func_185904_a();
+                        Material material1 = worldIn.func_180495_p(position.func_177981_b(k4)).func_185904_a();
 
                         if (material1 == Material.field_151579_a || material1 == Material.field_151584_j)
                         {
-                            this.func_175903_a(p_180709_1_, p_180709_3_.func_177981_b(k4), field_181645_a);
+                            this.func_175903_a(worldIn, position.func_177981_b(k4), field_181645_a);
                         }
                     }
 
